File: frontend/src/types/whatsapp.ts
export interface WhatsAppTemplate {
  key: string;
  title: string;
  description: string;
  category: string;
  raw_template: string;
  rendered_body: string;
}

export interface WhatsAppMessage {
  id: number;
  lead_id: number;
  sender_id: number;
  sender_name?: string;
  template_key?: string;
  recipient_phone: string;
  recipient_name?: string;
  message_body: string;
  status: 'SENT' | 'DELIVERED' | 'READ' | 'REPLIED';
  sent_at: string;
  delivered_at?: string;
  read_at?: string;
  replied_at?: string;
  metadata_json?: string;
  wa_link?: string;
}

export interface WhatsAppSendMessagePayload {
  lead_id: number;
  template_key?: string;
  recipient_phone: string;
  message_body: string;
  metadata?: Record<string, any>;
}

File: backend/app/schemas/whatsapp.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime

class WhatsAppTemplateResponse(BaseModel):
    key: str
    title: str
    description: str
    category: str
    raw_template: str
    rendered_body: str

class WhatsAppSendMessageRequest(BaseModel):
    lead_id: int
    template_key: Optional[str] = "CUSTOM"
    recipient_phone: str
    message_body: str
    metadata: Optional[Dict[str, Any]] = None

class WhatsAppMessageResponse(BaseModel):
    id: int
    lead_id: int
    sender_id: int
    sender_name: Optional[str] = None
    template_key: Optional[str] = None
    recipient_phone: str
    recipient_name: Optional[str] = None
    message_body: str
    status: str
    sent_at: datetime
    delivered_at: Optional[datetime] = None
    read_at: Optional[datetime] = None
    replied_at: Optional[datetime] = None
    metadata_json: Optional[str] = None
    wa_link: Optional[str] = None

    class Config:
        from_attributes = True

class WhatsAppStatusUpdateRequest(BaseModel):
    status: str # SENT, DELIVERED, READ, REPLIED
